go/chat/storage/storage.go

```diff
@@ -630,6 +646,87 @@ func (s *Storage) applyExpunge(ctx context.Context, convID chat1.ConversationID,
 	return &expunge, nil
 }
 
+func (s *Storage) handleEphemeralPurge(ctx context.Context, convID chat1.ConversationID,
+	uid gregor1.UID, metadata *chat1.ConvEphemeralMetadata) (*chat1.ConvEphemeralMetadata, Error) {
+
+	s.Debug(ctx, "handleEphemeralPurge convID: %v, uid: %v, metadata %v)", convID, uid, metadata)
+	// Noop if there is no metadata
+	if metadata == nil {
+		return nil, nil
+	}
+
+	mem, err := s.ephemeralTracker.getMetadata(ctx, convID, uid)
+	switch err.(type) {
+	case nil:
+		if mem.LastKtime >= metadata.LastKtime {
+			// No-op if the effect has already been applied locally
+			return nil, nil
+		}
+		// No shortcuts, fallthrough to apply.
+	case MissError:
+		// We have no memory, assume it needs to be applied
+	default:
+		return nil, err
+	}
+
+	maxMsgID, err := s.idtracker.getMaxMessageID(ctx, convID, uid)
+	if err != nil {
+		return nil, err
+	}
+	rc := NewInsatiableResultCollector() // collect all messages
+	err = s.engine.ReadMessages(ctx, rc, convID, uid, maxMsgID)
```

Comment:
@mmaxim i was thinking that as a later optimization we can send down the minimum messageID between the `MAX(ktime)` and `ktime is NULL` for a conv so we have a pointer in the conversation which we know there are no exploding messages behind -- would it make sense to add a `minMsgID` to the `ReadMessages` call so we don't have to traverse the entire conversation each purge?

Reply:
As discussed in person, I think this is a good idea. Hopefully the `ResultCollector` interface can help you make it happen.